### PR TITLE
ci: migration gha pr prerelease

### PR DESCRIPTION
**Pull request by jportasa**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows (keep old codes)
- [x] Squash commits because Joan Porta did not accept CLA

### CODE REVIEW

File: build/package/windows/nri-386-installer/Product.wxs [hunk 1]
```diff
@@ -59,37 +59,41 @@
             </Component>
         </DirectoryRef>
         <DirectoryRef Id="CustomPluginsBinFolder">
-            <Component Id="CMP_V1_CUSTOM_PLUGINS_BIN" Guid="e4509a7a-409e-4a38-8705-37de9cf0d72d" KeyPath="yes">
```
Comment (brushknight) [inline build/package/windows/nri-386-installer/Product.wxs:62]: Is it fine to change this GUID?

Comment (paologallinaharbur) [reply build/package/windows/nri-386-installer/Product.wxs]: Discussed offline, safe to change GUID

File: .github/workflows/push_pr.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Validate code
+        run: make ci/validate
+
+  snyk:
+    name: Run security checks via snyk
+    runs-on: ubuntu-20.04
+    env:
+        SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
```
Comment (roobre) [inline .github/workflows/push_pr.yml:34]:
```suggestion
        SNYK_TOKEN: ${{ secrets.COREINT_SNYK_TOKEN }}
```

File: .github/workflows/prerelease.yml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Validate code
+        run: make ci/validate
+
+  snyk:
+    name: Run security checks via snyk
+    runs-on: ubuntu-20.04
+    env:
+        SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
```
Comment (roobre) [inline .github/workflows/prerelease.yml:43]:
```suggestion
        SNYK_TOKEN: ${{ secrets.COREINT_SNYK_TOKEN }}
```

File: build/.goreleaser.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    formats:
+      - rpm
+    bindir: "/var/db/newrelic-infra/newrelic-integrations/bin"
+    files:
+      "couchbase-config.yml.sample": "/etc/newrelic-infra/integrations.d/couchbase-config.yml.sample"
+      "CHANGELOG.md": "/usr/share/doc/nri-couchbase/CHANGELOG.md"
+      "README.md":  "/usr/share/doc/nri-couchbase/README.md"
+      "LICENSE": "/usr/share/doc/nri-couchbase/LICENSE"
+    config_files:
+      "couchbase-definition.yml": "/var/db/newrelic-infra/newrelic-integrations/couchbase-definition.yml"
```
Comment (roobre) [inline build/.goreleaser.yml:82]:
```suggestion
```

**Review by roobre — CHANGES_REQUESTED** [review]

**Review by roobre — APPROVED** [review]
Nothing wrong seen here

**Pull request merged**